src/System.Linq.Expressions/src/System/Dynamic/Utils/ListArgumentProvider.cs

```diff
@@ -66,31 +72,48 @@ public void RemoveAt(int index)
 
         #region ICollection<T> Members
 
+        [ExcludeFromCodeCoverage] // Unreachable
         public void Add(T item)
         {
             throw ContractUtils.Unreachable;
         }
 
+        [ExcludeFromCodeCoverage] // Unreachable
         public void Clear()
         {
             throw ContractUtils.Unreachable;
         }
 
         public bool Contains(T item) => IndexOf(item) != -1;
 
-        public void CopyTo(T[] array, int arrayIndex)
+        public void CopyTo(T[] array, int index)
```

Comment:
`arrayIndex` better matches the `ICollection<T>.CopyTo()` definition, but this will be called through `ReadOnlyCollection.CopyTo()` which uses `index` in its definition, which makes this seem the preferable parameter name to use in exceptions, though the fact that it could be called via `IList<T>` or `ICollection<T>` means no name is perfect here.